### PR TITLE
[#1337] Update user guide and release notes.

### PR DESCRIPTION
**Pull request by kaniyan**
This PR updates the user guide and the release notes for the feature in #1337.

### CODE REVIEW

File: site/documentation/content/concepts/resource-limits.md [hunk 1]
```diff
@@ -18,5 +18,6 @@ The MQTT and AMQP protocol adapters keep the connections longer opened than thei
 
 Hono supports limiting the number of messages that devices and north bound applications of a tenant can publish to Hono during a given time interval. Before accepting any telemetry or event or command messages from devices or north bound applications, it is checked by the protocol adapters that if the message limit is exceeded or not. The incoming message is discarded if the limit is exceeded. 
 
-The default prometheus based implementation uses data volume as the factor to limit the messages. The data volume already consumed by a tenant over the given time interval is compared with the configured message limit before accepting any messages.
+The default prometheus based implementation uses data volume as the factor to limit the messages. The data volume already consumed by a tenant over the given time interval is compared with the configured message limit before accepting any messages. The default implementation supports two modes of message limits calculation namely `days` and `monthly`. For more details on how to set the mode refer to the [Tenant API]({{< relref "/api/tenant#data-volume-period-configuration-format" >}}).
```
Comment (sophokles73) [inline site/documentation/content/concepts/resource-limits.md]: The default *Prometheus* based ...

File: site/documentation/content/concepts/resource-limits.md [hunk 2]
```diff
@@ -18,5 +18,6 @@ The MQTT and AMQP protocol adapters keep the connections longer opened than thei
...
 
+In the `monthly` mode, the data usage is calculated from the beginning till the end of the current (Gregorian) calendar month and compared with the configured *maxBytes*. Only for the first month, the *effective maxBytes* are calculated based on the configured *maxBytes* and the remaining days in the month from the *effectiveSince* date. The data usage for those remaining days are then compared with the calculated *effective maxBytes*. In the `days` mode, the data usage is calculated for the number of days defined by the configuration property *no-of-days* and then compared with the *maxBytes*.
```
Comment (sophokles73) [inline site/documentation/content/concepts/resource-limits.md]: If this is supposed to be the main (and most exhaustive) source of information regarding how message limits work in Hono, then I think we will need to be a little more comprehensive here, e.g. providing some examples for how the max bytes are calculated in either case.

File: site/documentation/content/concepts/resource-limits.md (hunk 1 above)
Comment (kaniyan) [reply site/documentation/content/concepts/resource-limits.md]: Done.

File: site/documentation/content/concepts/resource-limits.md (hunk 2 above)
Comment (kaniyan) [reply site/documentation/content/concepts/resource-limits.md]: @sophokles73 I tried to explain more comprehensive with some examples. Would you mind taking a look?

**Pull request merged**